### PR TITLE
SUS-723 | Checking namespace before displaying wall message

### PR DESCRIPTION
**Pull request by rybmat**
https://wikia-inc.atlassian.net/browse/SUS-723

### CODE REVIEW

File: extensions/wikia/Wall/WallMessage.class.php [hunk 1]
```diff
@@ -63,7 +63,7 @@ static public function newFromId( int $id, $master = false ) {
 
 		$title = Title::newFromID( $id, $master == true ? Title::GAID_FOR_UPDATE : 0 );
```
Comment (WikiaTech) [inline extensions/wikia/Wall/WallMessage.class.php:66]: MINOR Remove the literal "true" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 46 issues
* MAJOR 17 major
* MINOR 17 minor
* INFO 12 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [WallMessage.class.php](https://github.com/Wikia/app/blob/ddc61a32f3a84864e1cdd649c01b89369ad43947/extensions/wikia/Wall/WallMessage.class.php): File "WallMessage.class.php" has 1772 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS104)
1. MAJOR [WallMessage.class.php#L7](https://github.com/Wikia/app/blob/ddc61a32f3a84864e1cdd649c01b89369ad43947/extensions/wikia/Wall/WallMessage.class.php#L7): Class "WallMessage" has 127 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. MAJOR [WallMessage.class.php#L50](https://github.com/Wikia/app/blob/ddc61a32f3a84864e1cdd649c01b89369ad43947/extensions/wikia/Wall/WallMessage.class.php#L50): Reduce the number of returns of this function 5, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MAJOR [WallMessage.class.php#L162](https://github.com/Wikia/app/blob/ddc61a32f3a84864e1cdd649c01b89369ad43947/extensions/wikia/Wall/WallMessage.class.php#L162): This function has 8 parameters, which is greater than the 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS107)
1. MAJOR [WallMessage.class.php#L162](https://github.com/Wikia/app/blob/ddc61a32f3a84864e1cdd649c01b89369ad43947/extensions/wikia/Wall/WallMessage.class.php#L162): Reduce the number of returns of this function 5, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MAJOR [WallMessage.class.php#L227](https://github.com/Wikia/app/blob/ddc61a32f3a84864e1cdd649c01b89369ad43947/extensions/wikia/Wall/WallMessage.class.php#L227): Remove this unused "$userMaster" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. MAJOR [WallMessage.class.php#L437](https://github.com/Wikia/app/blob/ddc61a32f3a84864e1cdd649c01b89369ad43947/extensions/wikia/Wall/WallMessage.class.php#L437): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MAJOR [WallMessage.class.php#L495](https://github.com/Wikia/app/blob/ddc61a32f3a84864e1cdd649c01b89369ad43947/extensions/wikia/Wall/WallMessage.class.php#L495): Reduce the number of returns of this function 7, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MAJOR [WallMessage.class.php#L530](https://github.com/Wikia/app/blob/ddc61a32f3a84864e1cdd649c01b89369ad43947/extensions/wikia/Wall/WallMessage.class.php#L530): Reduce the number of returns of this function 8, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MAJOR [WallMessage.class.php#L671](https://github.com/Wikia/app/blob/ddc61a32f3a84864e1cdd649c01b89369ad43947/extensions/wikia/Wall/WallMessage.class.php#L671): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)

**Review by macbre — APPROVED** [review]

**Comment by TK-999** [comment]
Should we log if something like this happens? IMO this is a definitive indicator of broken state of `comments_index` so we should make sure we have context on it when it happens.

**Review by mszabo-wikia — CHANGES_REQUESTED** [review]
Should we log if something like this happens? IMO this is a definitive indicator of broken state of comments_index so we should make sure we have context on it when it happens.

File: extensions/wikia/Wall/WallMessage.class.php [hunk 2]
```diff
@@ -131,6 +131,10 @@ static public function newFromIds( Array $ids ) {
 		return array_values( $wallMessages );
 	}
 
+	static private function isWallMessage( $title ) {
+		return in_array( $title->getNamespace(), [ NS_USER_WALL_MESSAGE, NS_USER_WALL_MESSAGE_GREETING ] );
```
Comment (mszabo-wikia) [inline extensions/wikia/Wall/WallMessage.class.php]: What about Forum?

**Comment by rybmat** [comment]
@TK-999 as mentioned in ticket's comment, when proposal for delete flow will be implemented and released this state will not be possible because entry from comments index will be removed when removing  the message from wall. Currently there is a lot of such broken entries in db and I plan to write maintenance script to remove them. Knowing that I don't think that logging is necessary

File: extensions/wikia/Wall/WallMessage.class.php (hunk 2 above)
Comment (rybmat) [reply extensions/wikia/Wall/WallMessage.class.php]: From what I have seen forum checks namespace, but if you know place where it should also be applied please give me a link ;)

**Review by mszabo-wikia — APPROVED** [review]

**Review by macbre — APPROVED** [review]

File: extensions/wikia/Wall/WallMessage.class.php [hunk 3]
```diff
@@ -133,6 +133,10 @@ static public function newFromIds( Array $ids ) {
 		return array_values( $wallMessages );
 	}
 
+	static public function isWallMessage( Title $title ) {
+		return in_array( $title->getNamespace(), [ NS_USER_WALL_MESSAGE, NS_USER_WALL_MESSAGE_GREETING, NS_WIKIA_FORUM_BOARD_THREAD ] );
```
Comment (mszabo-wikia) [inline extensions/wikia/Wall/WallMessage.class.php]: Let's use `Title::inNamespaces` helper here.

**Review by macbre — APPROVED** [review]

**Pull request merged**